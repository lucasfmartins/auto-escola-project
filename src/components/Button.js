import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components";

//#Com styled components, nÃ£o precisamos definir type e chieldren, por serem 1/1
// const Button = ({type, children}) => (
//   <button type={type}>
//     {children}
//   </button>
// );

const getPrimaryColor = (props) => props.theme.colors.primary.main;
const getPrimaryText = (props) => props.thme.colors.primary.text;

const Button = styled.button`
  font-size:  1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: ${getPrimaryColor};
  border: 2px solid ${getPrimaryColor};
  color: ${getPrimaryText};
`;

Button.defaultProps = {
  type: "button",
  children: undefined,
}

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node,
}

export default Button;
