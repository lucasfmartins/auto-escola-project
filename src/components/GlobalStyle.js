import {createGlobalStyle} from 'styled-components';

const GlobalStyle = createGlobalStyle`   
    @import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;600;700&display=swap');
    html{
        font-family: 'Poppins', sans-serif;
        font-weight: 300;
        font-size: 100%; //padrão 16px
        color: #212121;
        box-sizing: border-box; //para que os navegaores entendão que os elementos tem o tamanho do padding
    }

    *, *:before, *:after{
      box-sizing: inherit;
    }

    body {
      margin: 0;
    }
`;

export default GlobalStyle
