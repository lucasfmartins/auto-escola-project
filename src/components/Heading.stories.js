import React from 'react';
import { text } from "@storybook/addon-knobs";

import Heading from "./Heading";

export default {
  title: "Components/Heading",
  component: Heading
}

//função que retorna o proprio componente
//criando nossa historia

export const usage = () => (
  <div>
    <Heading><h1>{text("Título 1","Título 1")}</h1></Heading>

    <Heading><h2>{text("Título 2","Título 2")}</h2></Heading>

    <Heading><h3>{text("Título 3","Título 3")}</h3></Heading>

    <Heading><h4>{text("Título 4","Título 4")}</h4></Heading>

    <Heading><h5>{text("Título 5","Título 5")}</h5></Heading>

    <Heading><h6>{text("Título 6","Título 6")}</h6></Heading>

  </div>
  

);
