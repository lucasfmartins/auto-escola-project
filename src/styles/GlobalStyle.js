import {createGlobalStyle} from 'styled-components';

import {BreakpointSizes, BreakAt} from "./Breakpoints";

const GlobalStyle = createGlobalStyle`   
    @import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;600;700&display=swap');
    html{
        font-family: 'Poppins', sans-serif;
        font-weight: 300;
        font-size: 100%; //padrão 16px
        color: #212121;
        box-sizing: border-box; //para que os navegaores entendão que os elementos tem o tamanho do padding
    }

    *, *:before, *:after{
      box-sizing: inherit;
    }

    body {
      margin: 0;
    }

    h1, h2, h3, h4, h5, h6 {
      line-height: 1.3;
      font-weight: 700;
      letter-spacing: 2px;

      strong{
      color: #ffc107;
    }
    }

    h1{
      //mobile first
      font-size: 2.5rem; // 16 * 2.5 = 40px 

      ${BreakAt(BreakpointSizes.lg)}{
        font-size: 3.75rem;       
      }
    }

    h2 {
      font-size: 2rem;

      ${BreakAt(BreakpointSizes.lg)}{
        font-size: 3.125rem;
      }
    }

    h3 {
      font-size: 1.9rem;

      ${BreakAt(BreakpointSizes.lg)}{
        font-size: 2.5rem;
      }
    }

    h4 {
      font-size: 1.3rem;
      font-weight: 600;

      ${BreakAt(BreakpointSizes.lg)}{
        font-size: 2.125rem;
      }
    }

    h5 {
      font-size: 1.2rem;
      font-weight: 300;

      ${BreakAt(BreakpointSizes.lg)}{
        font-size: 1.5rem;
      }
    }

    
    h6 {
      font-size: 1.1rem;
      font-weight: 600;

        ${BreakAt(BreakpointSizes.lg)}{
          font-size: 1.25rem;
      }
    }
`;


export default GlobalStyle
