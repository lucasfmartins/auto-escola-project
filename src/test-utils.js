import React from "react";

import ThemeProviderComponent from "styles/ThemeProvider";
import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";

const TestProvider = ({ children }) => (
  <Router>
    <ThemeProviderComponent>{children}</ThemeProviderComponent>
  </Router>
);

export * from "@testing-library/react";

//função que retonra o proprio render
//ui = component
// wrapper, passa o componente de provider, que faz o encapsulamento automatico
const customRender = (ui, options) =>
  render(ui, { wrapper: TestProvider, ...options });

//sobrescrevendo a importação do render
export { customRender as render };
