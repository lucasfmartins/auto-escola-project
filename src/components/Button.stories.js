import React from 'react';
import Button from './Button';
  
export default {
  title: "Components/Button",
  component: Button,
}


export const usage = () => <Button>Meu Botão</Button>;

  

