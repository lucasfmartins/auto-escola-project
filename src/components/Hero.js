import React from 'react';
import styled, {css} from 'styled-components';
import PropTypes from 'prop-types';

import {BreakAt, BreakpointSizes} from "styles/Breakpoints";

const colorYellow = "#ffc107";
const Root = styled.div`
    color: #fff;
    padding: 100px 0;
    ${props => css`
        background: url(${props.image}), rgba(0,0,0,0.4);
        background-size: cover;
        background-position: center;
        background-blend-mode: overlay;
        
    `}
`;

const Content = styled.div`
    p,
    li {
        font-size: 20px;
        font-weight: 300;
    }

    ul {
        list-style: none; 
        padding-left: 0;
    }

    li{     
        &::before{
          content: "\\2713\\0020";
          color: ${colorYellow};
        }
    }
`;

const Container = styled.div`
  /* background-color: red; */
  //menor tela, fica definido
  width: 100%;
  padding: 0 8px;

  // cria media querry para maior tela
  /* @media (min-width: 768){
    max-width: 1140px;
    margin: 0 16px; //em cima e embaix sem margem, e automatico dos lados
  } */
  ${BreakAt(BreakpointSizes.sm)}{
    padding:0 16px;
  }

  // cria media querry para tela ainda maior
  ${BreakAt(BreakpointSizes.lg)}{
    max-width: 1140px;
    margin: 0 auto; //em cima e embaix sem margem, e automatico dos lados
    padding: 0;
  }
  
`;

//children permite composição de componentes, 
const Hero = ({image, children}) => (
    <Root image={image} data-testid = "hero">
        <Container>
            <Content>{children}</Content>       
        </Container>
    </Root>
);

Hero.propTypes = {
  /**
   *Background 
   */
  image: PropTypes.string,
  children: PropTypes.node,
}

// Hero.defaultProps = {
//   title: "Meu título",
// }

export default Hero;

//children para compor os componentes

{/* <ul>
<li>Lorem ipsum dolor sit amet consectetur, adipisicing elit.</li>
<li>Lorem ipsum dolor sit amet consectetur, adipisicing elit.</li>
<li>Lorem ipsum dolor sit amet consectetur, adipisicing elit.</li>
<button>Matricule-se agora</button>
</ul> */}
