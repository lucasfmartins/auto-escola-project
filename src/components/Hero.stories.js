import React from "react";
import { text } from "@storybook/addon-knobs";

import Hero from "./Hero";
import BgHighWayImage from "../stories/assets/highway.jpg";
import BgCarImage from "../stories/assets/car.jpg";
import Heading from "./Heading";
import Button from "./Button";

export default {
  title: "Components/Hero",
  component: Hero,
};

export const usage = () => (
  <Hero image={BgHighWayImage}>
    <h1> {text("Title", "Ganhe sua liberdade para ir e vir")}</h1>
    <p> {text("Text", "A autoescola lider em aprovação")}</p>
  </Hero>
);

export const withList = () => (
  <Hero image={BgCarImage}>
    <Heading>
      <h1>
        Ganhe sua <strong>liberdade</strong>
        <br />
        para ir e vir
      </h1>
    </Heading>
    <ul>
      <li>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</li>
      <li>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</li>
      <li>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</li>
    </ul>
    <Button color="primary" variant="outlined">
      Matricule-se agora
    </Button>
  </Hero>
);

// ou usar () => return {}
