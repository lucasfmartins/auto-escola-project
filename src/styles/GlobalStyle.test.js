import React from 'react';
import { render } from "../components/test-utils";


import GlobalStyle from "./GlobalStyle";

test("match snapshot", () => {
  render (<GlobalStyle />);

  expect(document.head).toMatchSnapshot();
});

