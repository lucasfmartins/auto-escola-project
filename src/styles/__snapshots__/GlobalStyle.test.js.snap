// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

<head>
  <style
    data-styled="active"
    data-styled-version="5.2.1"
  >
    html{font-family:'Poppins',sans-serif;font-weight:300;font-size:100%;color:#212121;box-sizing:border-box;}
    *,*:before,*:after{box-sizing:inherit;}
    body{margin:0;}
    h1,h2,h3,h4,h5,h6{line-height:1.3;font-weight:700;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;}
    h1 strong,h2 strong,h3 strong,h4 strong,h5 strong,h6 strong{color:#ffc107;}
    h1{font-size:2.5rem;}
    @media (min-width:992px){h1{font-size:3.75rem;}}
    h2{font-size:2rem;}
    @media (min-width:992px){h2{font-size:3.125rem;}}
    h3{font-size:1.9rem;}
    @media (min-width:992px){h3{font-size:2.5rem;}}
    h4{font-size:1.3rem;font-weight:600;}
    @media (min-width:992px){h4{font-size:2.125rem;}}
    h5{font-size:1.2rem;font-weight:300;}
    @media (min-width:992px){h5{font-size:1.5rem;}}
    h6{font-size:1.1rem;font-weight:600;}
    @media (min-width:992px){h6{font-size:1.25rem;}}
  </style>
</head>
`;
